### PR TITLE
chore(DTFS2-4533): remove moment from gef-ui, tfm-api, tfm-ui

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
Moment js is deprecated, it should be replaced. This PR will not cover removing moment from tests - see #2761 .

## Resolution :heavy_check_mark:
* Replace moment() with vanilla Dates where possible and date-fns where necessary in
  * gef-ui 
  * trade-finance-manager-api
  * trade-finance-manager-ui

## Misc
* add `api-test-dev` scripts to all services
* update test config and linting for ts files in gef-ui

### CODE REVIEW

File: gef-ui/templates/includes/application-preview/task-comments.njk [hunk 1]
```diff
@@ -61,7 +61,7 @@
   <div class="govuk-!-margin-top-3" data-cy="latest-comment">
     {% for comment in comments %}
       <strong>Comments from {{ comment.firstname }} {{ comment.surname }}</strong>
-      <p>{{ comment.createdAt | localiseTimestamp('DD MMM YYYY h:mma', timezone) }}</p>
+      <p>{{ comment.createdAt | localiseTimestamp('dd MMM yyyy h:mmaaa', timezone) }}</p>
```
Comment (MarRobSoftwire) [inline gef-ui/templates/includes/application-preview/task-comments.njk:64]: Sanity check please:
|---------|--------|
|[before](https://momentjscom.readthedocs.io/en/latest/moment/04-displaying/01-format/) |  [after](https://date-fns.org/v3.3.1/docs/format)|
|`h` | `h`|
|`mm`| `mm`|
|`a`|`aaa`|

File: gef-ui/server/utils/display-items.js [hunk 2]
```diff
@@ -82,20 +82,20 @@ const facilityItems = (facilityUrl, {
...
       href: `${facilityUrl}?status=change`,
-      method: (callback) => (isTrueSet(callback) ? STAGE.ISSUED : STAGE.UNISSUED),
+      method: (value) => (isTrueSet(value) ? STAGE.ISSUED : STAGE.UNISSUED),
     },
     {
       label: 'Date issued to exporter',
       id: 'issueDate',
-      method: (callback) => moment(callback)
+      method: (value) => moment(value)
         .format('D MMMM YYYY'),
```
Comment (MarRobSoftwire) [inline gef-ui/server/utils/display-items.js:91]: Sanity check please:
|[moment](https://momentjscom.readthedocs.io/en/latest/moment/04-displaying/01-format/) |  [date-fns](https://date-fns.org/v3.3.1/docs/format)| example|
|---------|--------|---|
|`D` | `d`| 1, 2, 3, ... 31 |
|`MMMM`| `MMMM`| January |
|`YYYY`|`yyyy`| 2024 |

File: gef-ui/server/utils/display-items.js [hunk 3]
```diff
@@ -82,20 +82,20 @@ const facilityItems = (facilityUrl, {
...
       id: 'hasBeenIssued',
       href: `${facilityUrl}?status=change`,
-      method: (callback) => (isTrueSet(callback) ? STAGE.ISSUED : STAGE.UNISSUED),
+      method: (value) => (isTrueSet(value) ? STAGE.ISSUED : STAGE.UNISSUED),
     },
     {
       label: 'Date issued to exporter',
       id: 'issueDate',
-      method: (callback) => moment(callback)
+      method: (value) => moment(value)
```
Comment (MarRobSoftwire) [inline gef-ui/server/utils/display-items.js:90]: Need to confirm the type of value

File: trade-finance-manager-api/src/utils/date.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * @param dateStr an ISO-8601 (or other) date string
+ * @returns date formatted as `yyyy-MM-dd` or 'Invalid date' if not a valid format
+ */
+export const formatDate = (dateStr: string) => {
+  const date = new Date(dateStr);
+
+  return isValid(date) ? format(date, 'yyyy-MM-dd') : 'Invalid date';
+};
+
```
Comment (MarRobSoftwire) [inline trade-finance-manager-api/src/utils/date.ts:31]: Genuinely no idea how to replicate this with date-fns.
The unit tests want the input `'20210419'` to be converted to `'2021-04-19'`
Haven't investigated how this is used across the code yet but probably should just look to remove this function entirely

File: trade-finance-manager-api/src/utils/date.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ */
+// TODO: DTFS2-6998: remove this function
+export const formatDate = (dateStr: string) => moment(dateStr).format('YYYY-MM-DD');
+
+/**
+ * @param dateStr Unix timestamp representing number of milliseconds between this date and 1st January 1970 (UTC),
+ * stored as a string 
+ * @returns date formatted as `yyyy-MM-dd` or 'Invalid date' if not a number
+ */
+export const formatTimestamp = (dateStr: string) => {
```
Comment (MarRobSoftwire) [inline trade-finance-manager-api/src/utils/date.ts:35]: Note to review whether any of these functions are still needed

File: gef-ui/api-test.jest.config.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
+  preset: 'ts-jest/presets/js-with-babel',
```
Comment (toddym42) [inline gef-ui/api-test.jest.config.js:2]: Elsewhere we've done
```
preset: 'ts-jest',
transform: {
  '^.+\\.js$': 'babel-jest',
  '^.+\\.ts$': 'ts-jest',
}
```
Is this a shorthand for that?

**Pull request closed without merging**